src/main/java/frc/robot/subsystems/swerve/swervestatehelpers/SwerveStateHelper.java

```diff
@@ -0,0 +1,118 @@
+import org.littletonrobotics.junction.Logger;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static frc.robot.subsystems.swerve.swervestatehelpers.AimAssistUtils.getRotationAssistedSpeeds;
+
+public class SwerveStateHelper {
+
+    private final Swerve swerve;
+    private final SwerveConstants swerveConstants;
+    private final Supplier<Pose2d> robotPoseSupplier;
+    private final Supplier<Optional<Translation2d>> noteTranslationSupplier;
+
+    public SwerveStateHelper(Supplier<Pose2d> robotPoseSupplier, Supplier<Optional<Translation2d>> noteTranslationSupplier,
+            Swerve swerve) {
+        this.swerve = swerve;
+        this.swerveConstants = swerve.getConstants();
+        this.robotPoseSupplier = robotPoseSupplier;
+        this.noteTranslationSupplier = noteTranslationSupplier;
+    }
+
+    public ChassisSpeeds applyAimAssistOnSpeeds(AimAssist aimAssistState, ChassisSpeeds chassisSpeeds) {
+        return switch (aimAssistState) {
+            case SPEAKER -> getRotationAssistedSpeeds(
+                    chassisSpeeds,
+                    robotPoseSupplier.get().getRotation(),
+                    SwerveMath.getRelativeTranslation(robotPoseSupplier.get().getTranslation(), new Translation2d(0,0)).getAngle(),
+                    swerveConstants
+            );
+            case AMP -> getRotationAssistedSpeeds(
+                    chassisSpeeds,
+                    robotPoseSupplier.get().getRotation(),
+                    Field.getAngleToAmp(),
+                    swerveConstants
+            );
+            case NOTE -> handleNoteAssistState(
+                    chassisSpeeds,
+                    robotPoseSupplier,
+                    noteTranslationSupplier
+            );
+            case NONE -> chassisSpeeds;
+        };
+    }
+
+
+    private ChassisSpeeds handleNoteAssistState(
+            ChassisSpeeds speeds,
+            Supplier<Pose2d> robotPoseSupplier,
+            Supplier<Optional<Translation2d>> noteTranslationSupplier) {
+        if (noteTranslationSupplier.get().isEmpty()) {
+            return speeds;
+        }
+
+        Translation2d noteRelativeToRobot = SwerveMath.getRelativeTranslation(robotPoseSupplier.get(),
+                noteTranslationSupplier.get().get());
+
+
+        double pidGainHorizontalVelocity = swerveConstants.yMetersPIDController().calculate(
+                0,
+                noteRelativeToRobot.getY()
+        );
+
+        Logger.recordOutput("current angle", robotPoseSupplier.get().getRotation().getDegrees());
+        Logger.recordOutput("target note", noteTranslationSupplier.get().get());
+        return new ChassisSpeeds(
+                speeds.vxMetersPerSecond + pidGainHorizontalVelocity * Math.sin(robotPoseSupplier.get().getRotation().unaryMinus().getRadians()),
+                speeds.vyMetersPerSecond + pidGainHorizontalVelocity * Math.cos(robotPoseSupplier.get().getRotation().unaryMinus().getRadians()),
```

Comment:
pidGainHorizontalVelocity * Math.sin(robotPoseSupplier.get().getRotation().unaryMinus().getRadians())
pidGainHorizontalVelocity * Math.cos(robotPoseSupplier.get().getRotation().unaryMinus().getRadians())

put these two in vars